CMakeLists.txt

```diff
@@ -2,6 +2,13 @@ cmake_minimum_required(VERSION 3.15)
 
 project(mender)
 
+# Set the default build type to MinSizeRel. We care more about size than speed.
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE MinSizeRel CACHE STRING
+    "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel."
```

Comment:
Shouldn't `Release` be changed to yield size-optimized binaries instead?

Reply:
Why? `MinSizeRel` is literally a minimum sized release. Which is exactly what we want.

Reply:
And when I say release, I mean `Release`

Reply:
I agree in that optimizing `MinSizeRel` is more correct. A user can always set `CMAKE_BUILD_TYPE` to Release and then I think it is expected to not tamper with these optimizations.

Reply:
To me, release binaries and packages should be built with the `Release` build type. And its parameters should determine how the released binaries/packages look like. This also gives more flexibility, take the `-Oz` flag for example -- it makes total sense for `MinSizeRel`, but `Release` may not use it because it's too aggressive. I guess this gets us to the broader discussion below, though.